src/hmr-server-engine.ts

```diff
@@ -38,19 +51,17 @@ export class EsmHmrEngine {
     return null;
   }
 
-  setEntry(sourceUrl: string, imports: string[], isHmrEnabled: boolean) {
+  setEntry(sourceUrl: string, imports: string[]) {
     const result = this.getEntry(sourceUrl, true)!;
     const outdatedDependencies = new Set(result.dependencies);
-    result.isHmrEnabled = isHmrEnabled;
-
     for (const importUrl of imports) {
       this.addRelationship(sourceUrl, importUrl);
       outdatedDependencies.delete(importUrl);
     }
 
-    for (const importUrl of outdatedDependencies) {
```

Comment:
lets not mix patterns here, since we use a `for of` immediately above

Reply:
for of shouldn't work on sets afaik (linter warns for it)

Reply:
I think that's a stylistic warning from your linter, and not a technical limitation. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
All right will revert both then